test/src/e2e_vm_tests/test_programs/should_fail/array_wrong_elements_types/src/main.sw

```diff
@@ -0,0 +1,45 @@
+script;
+
+fn vec<T>() -> Vec<T> {
+    Vec::new()
+}
+
+fn main() {
+    // unexpected u16
+    let a: [u8;1] = [1u16];
+
+    // unexpected u16
+    let _ = [1u8, 1u16];
+    let a = [1, 2u8, 3u16, 4u32, 5u64];
+
+    // unexpected u8
+    let _ = [1, 1u16, a[0]];
+
+    // unexpected string slice
+    let _ = [1, "", 1u8, 1u16];
+
+    // unexpected u8
+    let _ = [return, 1u8, 1u16];
+
+    // unexpected u16
+    let _ = [1u8, return, 1u16];
+
+    // unexpected u16
+    let _ = [1, return, 1u8, 1u16];
+
+    // unexpected str
+    let _ = [1, "", 1u16];
+    let a = [1, 2, "hello"];
+    let _ = [1, return, "", 1u16];
```

Comment:
On principle I would also add

```
let _ = [1, "", return, 1u16];
```

It ought to do the same as the case in line 19, but we treat `Never` as a special case, so it might be worth it just in case.

Reply:
Done.